### PR TITLE
Update OTel exporter changelog

### PR DESCRIPTION
**Pull request by jeanbisutti**

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
+AzureMonitorExporter.customize(sdkBuilder);
+OpenTelemetry openTelemetry = sdkBuilder.build().getOpenTelemetrySdk();
+```
+
 ### Bugs Fixed
+- [Fix a race condition RejectedExecutionException](https://github.com/Azure/azure-sdk-for-java/pull/41927)
 
 ### Other Changes
+- [Better populate exception type name and message from span events](https://github.com/Azure/azure-sdk-for-java/pull/41397)
+- [Update OpenTelemetry to 1.42.1]([https://github.com/Azure/azure-sdk-for-java/pull/39843](https://github.com/Azure/azure-sdk-for-java/pull/41931))
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md:22]:
```suggestion
- [Update OpenTelemetry to 2.8.0](https://github.com/Azure/azure-sdk-for-java/pull/41931))
```

**Comment by heyams** [comment]
let's include this https://github.com/Azure/azure-sdk-for-java/pull/42134 too

File: sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md (hunk 1 above)
Comment (trask) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md:22]: The API/SDK version (1.42.1) is probably most relevant to this repository.

Probably good to be more explicit:

```suggestion
- [Update OpenTelemetry SDK to 1.42.1]([https://github.com/Azure/azure-sdk-for-java/pull/39843](https://github.com/Azure/azure-sdk-for-java/pull/41931))
```

Comment (heyams) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md:22]: i'm fine with it.. it doesn't need to mention #39843. that has nothing to do with 2.8 upgrade.

**Comment by jeanbisutti** [comment]
> Need to mention [major API change](https://github.com/Azure/azure-sdk-for-java/pull/41705) in this changelog.

See the breaking changes part of the change log

File: sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md [hunk 2]
```diff
@@ -193,7 +208,7 @@
 
 ### New Features
 - `AzureMonitorExporterBuilder` now supports reading connection string from `APPLICATIONINSIGHTS_CONNECTION_STRING
-` environment variable.
+  ` environment variable.
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md:211]: can you reformat these 2 lines into one line. an extra space?

File: sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
+
 ### Bugs Fixed
-- Fix a bug where post requests to Live Metrics returned http 400 due to malformed json.
+- [Fix a race condition RejectedExecutionException](https://github.com/Azure/azure-sdk-for-java/pull/41927)
+- [Fix a bug where post requests to Live Metrics returned http 400 due to malformed json](https://github.com/Azure/azure-sdk-for-java/pull/42147)
 
 ### Other Changes
+- [Better populate exception type name and message from span events](https://github.com/Azure/azure-sdk-for-java/pull/41397)
+- [Update OpenTelemetry SDK to 1.42.1](https://github.com/Azure/azure-sdk-for-java/pull/41931))
+- [Improve the error message when non connection string](https://github.com/Azure/azure-sdk-for-java/pull/42034)
```
Comment (heyams) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/CHANGELOG.md:26]:
```suggestion
- [Improve the error message for null connection string](https://github.com/Azure/azure-sdk-for-java/pull/42034)
```

**Review by trask — APPROVED** [review]

**Comment by jeanbisutti** [comment]
> /check-enforcer override

See https://github.com/Azure/azure-sdk-for-java/pull/42163

**Pull request merged**